src/Our.Umbraco.InnerContent/Web/UI/App_Plugins/InnerContent/views/innercontent.doctypepicker.html

```diff
@@ -1,42 +1,38 @@
 ﻿<div class="inner-content__doctypepicker" ng-controller="Our.Umbraco.InnerContent.Controllers.DocTypePickerController as vm">
-    <div>
-        <table class="table table-striped">
-            <thead>
-                <tr>
-                    <th class="td-min" />
-                    <th>
-                        Document Type
-                    </th>
-                    <th>
-                        Name Template
-                        <i class="icon icon-help-alt inner-content__help-icon"
-                           data-tooltip="Enter an angularjs expression to evaluate against each item for its name."
-                           ng-mouseover="vm.tooltipMouseOver($event)"
-                           ng-mouseleave="vm.tooltipMouseLeave()"></i>
-                    </th>
-                    <th class="td-min" ng-show="model.value.length > 1" />
-                </tr>
-            </thead>
-            <tbody ui-sortable="vm.sortableOptions" ng-model="model.value">
-                <tr ng-repeat="config in model.value">
-                    <td class="icon icon-navigation"></td>
-                    <td>
-                        <select ng-options="dt.guid as dt.name for dt in vm.docTypes"
-                                ng-model="config.icContentTypeGuid" required></select>
-                    </td>
-                    <td>
-                        <input type="text" ng-model="config.nameTemplate" />
-                    </td>
-                    <td ng-show="model.value.length > 1">
-                        <a class="btn btn-danger" ng-click="vm.remove($index)">Remove</a>
-                    </td>
-                </tr>
-            </tbody>
-        </table>
-        <div>
-            <a href="" class="btn" ng-click="vm.add()">Add</a>
+    <div class="inner-content__help-container">
+        <i class="icon icon-help-alt inner-content__help-icon"
+           data-tooltip="Name Template: Enter an angularjs expression to evaluate against each item for its name."
+           ng-mouseover="vm.tooltipMouseOver($event)"
+           ng-mouseleave="vm.tooltipMouseLeave()"></i>
+    </div>
+    <div class="list-view-layouts" ui-sortable="vm.sortableOptions" ng-model="model.value">
+        <div class="list-view-layout" ng-repeat="config in model.value">
+            <i class="icon-navigation list-view-layout__sort-handle"></i>
+            <div>
+                <a href="" ng-click="vm.openDocTypePicker(config)" class="list-view-layout__icon">
+                    <i class="{{ vm.getContentType(config.icContentTypeGuid).icon ||  'icon-stop' }}"></i>
```

Comment:
This might be something or nothing... I'm wondering about the multiple calls to `vm.getContentType`, if there's a point when it slows down.

I've tried this with 29 doctypes, and it performs fine (on my machine; Surfacebook 1).

@dnwhte How many doctypes are you working with?

Reply:
I've only tested with one of the starter kits (maybe around 20 doctypes). I imagine at some point it could cause some issues. My angular skills aren't that great, but I'm pretty sure each call triggers a new digest cycle which might cause ng-repeat to iterate the collection multiple times?

I considered two other options, but thought they might be overkill or the PR:
1. Change what is stored in the model, so it includes the required fields (icon, name, alias)
2. Create another array with the required fields for iterating in the view. This would require doing some sort of on change watch on the array to update the model when necessary.

Reply:
Fixed. Went with option 2 - https://github.com/umco/umbraco-inner-content/pull/45/commits/3421829a0895fc3ca1e3af2bc128cb1440d63a02